### PR TITLE
feat(search): Add search for field level description and tags

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
Add search for field level description and tags. Also add search for dataset description. 

For field-level description and tags, we have two sources - one from ingest one from UI (editable). We simply add multiple search fields instead of merging them. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/metadata/search/DatasetDocument.pdl
Comment (jywadhwani) [inline metadata-models/src/main/pegasus/com/linkedin/metadata/search/DatasetDocument.pdl:89]: My understanding is that `fieldDescriptions` is coming from `SchemaMetadata`, whereas `editedFieldDescriptions` is coming from `EditableSchemaMetadata`. But what is the source of truth when it comes to field description? If a user has made changes to a field description via UI, shouldn't that be the "updated" field description? Why do we need to keep both?
Not sure if this has been discussed before. If yes, please feel free to point to relevant doc/rfc.

**Review by shirshanka — APPROVED** [review]
LGTM!

File: metadata-models/src/main/pegasus/com/linkedin/metadata/search/DatasetDocument.pdl
Comment (shirshanka) [reply metadata-models/src/main/pegasus/com/linkedin/metadata/search/DatasetDocument.pdl:89]: That's a good question @jywadhwani 
I'm inclined to let this implementation stay for the following reasons: 
1. We should keep the "human written" documentation separate from the "schema-attached" documentation (we are already doing that in the db, so makes sense to continue doing it in the indexes. This way, we don't mix the two streams of information (especially without a governed approval flow for documentation overrides). 
2. Once we build that sort of approval flow in the application, we can store breadcrumbs for when an editable schema field description overrides an existing schema field description (imagine a flag on the schema-field-description marking it as not-to-be-used), which would trigger the indexer to remove this text from the index. 

Hope this addresses your concern.

Comment (jywadhwani) [reply metadata-models/src/main/pegasus/com/linkedin/metadata/search/DatasetDocument.pdl:89]: Sounds good, thanks for the explanation @shirshanka.

**Pull request merged**
